platforms/ide/problems/src/main/java/org/gradle/problems/internal/services/DefaultProblemSummarizer.java

```diff
@@ -99,8 +100,10 @@ public void emit(Problem problem, @Nullable OperationIdentifier id) {
     }
 
     private boolean exceededThreshold(Problem problem) {
-        ProblemId problemId = problem.getDefinition().getId();
-        AtomicInteger count = seenProblemsWithCounts.computeIfAbsent(problemId, key -> new AtomicInteger(0));
-        return count.incrementAndGet() > threshold;
+        int count = seenProblemsWithCounts.compute(
+            problem.getDefinition().getId(),
+            (key, value) -> value == null ? 1 : value + 1
+        );
```

Comment:
💅  Ah, I thought there was something, but I didn't remember initially. Since java 8 there is a very neat way of [using `Map.merge`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ConcurrentMap.html#merge(K,V,java.util.function.BiFunction)) to express counting maps:
```
myMap.merge(key, 1, Integer::sum)
```

For `ConcurrentHashMap` it is also a safe atomic operation.
See this [StackOverflow answer](https://stackoverflow.com/a/42648785) for the explanation.